src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -435,24 +450,39 @@ export class BundleDataClient {
     const hubPoolClient = this.clients.hubPoolClient;
     // Determine which bundle we should fetch from arweave, either the pending bundle or the latest
     // executed one. Both should have arweave data but if for some reason the arweave data is missing,
-    // this function will have to compute the bundle data from scratch which will be slow. We have to fallback
-    // to computing the bundle from scratch since this function needs to return the full bundle data so that
-    // it can be used to get the running balance proposed using its data.
-    const bundleBlockRanges = getImpliedBundleBlockRanges(
+    // this function will load the bundle data from the most recent bundle data published to Arweave.
+
+    let bundleBlockRanges = getImpliedBundleBlockRanges(
       hubPoolClient,
       this.clients.configStoreClient,
       hubPoolClient.hasPendingProposal()
         ? hubPoolClient.getLatestProposedRootBundle()
-        : hubPoolClient.getLatestFullyExecutedRootBundle(hubPoolClient.latestBlockSearched)! // ! because we know there is a bundle
+        : hubPoolClient.getNthFullyExecutedRootBundle(-1)!
     );
-    return {
-      blockRanges: bundleBlockRanges,
-      bundleData: await this.loadData(
-        bundleBlockRanges,
-        this.spokePoolClients,
-        true // this bundle data should have been published to arweave
-      ),
-    };
+    // Check if bundle data exists on arweave, otherwise fallback to last published bundle data. If the
+    // first bundle block range we are trying is the pending proposal, then we'll grab the most recently
+    // validated bundle, otherwise we'll grab the second most recently validated bundle.
+    let n = hubPoolClient.hasPendingProposal() ? 1 : 2;
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```

Comment:
```suggestion
    while (n++ < 4) {
```

Reply:
wdyt of faebdb8a8020235f85c295d254ea2dd32536c3b4 where we fallback to just loading from scratch.